AK/StringFloatingPointConversions.cpp

```diff
@@ -0,0 +1,1126 @@
+/*
+ * Copyright (c) 2022, Dan Klishch <danilklishch@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Array.h>
+#include <AK/BuiltinWrappers.h>
+#include <AK/FloatingPoint.h>
+#include <AK/StringFloatingPointConversions.h>
+#include <AK/UFixedBigInt.h>
+
+namespace AK {
+
+// This entire algorithm is an implementation of the paper: Ryu: Fast Float-to-String Conversion
+// by Ulf Adams, available at https://dl.acm.org/doi/pdf/10.1145/3192366.3192369 and an implemenetation
+// at https://github.com/ulfjack/ryu . A lot of possible mistakes from the article were corrected, see
+// discussion at https://github.com/SerenityOS/serenity/pull/15796 .
+//
+// Not implemented for float80, as it will require an insane lookup table size (193Kb).
+//
+// Run stress tests from https://github.com/DanShaders/serenity-arithmetic-benchmark after non-trivial
+// modifications.
+
+// These approximations should match the ones used in the Python script.
+static constexpr i64 log10_5_num = 10043;
+static constexpr i64 log10_5_denum = 14369;
+
+static constexpr i64 log10_2_num = 1406;
+static constexpr i64 log10_2_denum = 4671;
+
+static constexpr i64 log2_5_num = 8245;
+static constexpr i64 log2_5_denum = 3551;
+
+template<typename Number, size_t Size1, size_t Size2>
+struct LookupInformation {
+    i32 b0, b1; // B0 and B1 from the paper (accidentally swapped)
+    Number lt[Size1];
+    Number ge[Size2];
+};
+
+template<FloatingPoint>
+int lookup_table;
+
+template<typename FloatingPoint, typename MultiplyAndShiftFunction>
+FloatingPointExponentialForm inner_convert_floating_point_to_decimal_exponential_form(FloatingPoint value, MultiplyAndShiftFunction const& multiply_and_shift)
+{
+    using Extractor = FloatExtractor<FloatingPoint>;
+
+    Extractor bit_representation { .d = value };
+
+    bool sign = bit_representation.sign;
+    i32 exponent = bit_representation.exponent;
+    u64 mantissa = bit_representation.mantissa;
+
+    // For +0, it is {.sign = 0, fraction = 0, exponent = 0},
+    // for -0, is {.sign = 1, fraction = 0, exponent = 0},
+    if (exponent == 0 && mantissa == 0)
+        return { sign, 0, 0 };
+
+    // for +inf, -inf, and NaN is undefined.
+    // To be on the safe side, return something meaningless rather than just not checking.
+    if (exponent == Extractor::exponent_max) [[unlikely]]
+        return { 0, 0, 0 };
+
+    // Step 1. Decode the floating point number, and unify normalized and subnormal cases.
+    u64 real_mantissa = (exponent == 0 ? 0 : (1ull << Extractor::mantissa_bits)) + mantissa;
+    i32 real_exponent = (exponent == 0 ? 1 : exponent) - Extractor::exponent_bias - Extractor::mantissa_bits;
+    // abs(value) = real_mantissa * 2 ^ real_exponent
+
+    // Step 2. Determine the interval of information-preserving outputs.
+    // u, v, w are, respectively, lower bound for answer, exact value and upper bound for answer.
```

Comment:
I still think that if `u`, `v` and `w` have a more meaning with bounds they should have those names but with the comments of the algorithm added this is already more understandable. 
So you don't have to change this.